### PR TITLE
Plug hole that allowed concurrent replication streams

### PR DESCRIPTION
**Pull request by kirg**
- fix for concurrent replication streams
- update the 'first seqnum' in replication code-path (this was missing)
- added more logs around open/close extent

### CODE REVIEW

File: services/storehost/extMgr.go [hunk 1]
```diff
@@ -363,12 +372,17 @@ func (xMgr *ExtentManager) closeExtent(ext *extentContext, intent OpenIntent, op
...
 		ext.extMetrics.Decrement(load.ExtentMetricNumReadConns)
 		xMgr.hostMetrics.Decrement(load.HostMetricNumReadConns)
+
 	case OpenIntentReplicateExtent:
-		// mark as not open for replication
-		atomic.StoreInt32(&ext.openedForReplication, 0)
+
+		if openError != nil {
+			// mark as not open for replication
+			atomic.StoreInt32(&ext.openedForReplication, 0)
```
Comment (kirg) [inline services/storehost/extMgr.go]: ^^ the fix

Comment (thuningxu) [reply services/storehost/extMgr.go]: if openError == nil?

**Comment by thuningxu** [comment]
Can you add unit test?

**Comment by kirg** [comment]
There is a unit-test that checks to see that when two OpenReplicateStream requests come in, only one will succeed and the other will fail. Unfortunately, this bug shows up when a third OpenReplicateStream request comes in!

**Comment by kirg** [comment]
@thuningxu I have updated the existing test to try another (third) replicate-extent request, which should fail; verified that it fails without fix and passes with.

**Pull request merged**